src/unistd.rs

```diff
@@ -2522,30 +2521,29 @@ pub enum UserQuery {
     ///
     /// Internally, this function calls
     /// [getpwnam_r(3)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/getpwuid_r.html)
-    Name(String),
+    Name(&'a str),
     UidWithBufsize(Uid, usize),
-    NameWithBufsize(String, usize)
+    NameWithBufsize(&'a str, usize)
 }
 
 /// Query a group.
 ///
 /// The buffer size variants should not be needed 99% of the time; the default buffer size of 1024
-/// should be more than adequate. Only use the buffer size variants if you receive an ERANGE error
+/// should be more than adequate. Only use the buffer size variants if you receive an `ERANGE` error
 /// without them, or you know that you will be likely to.
 ///
 /// # Examples
 ///
-/// ```ignore
-/// use nix::unistd::{Group, GroupQuery};
-/// // On many systems there's no `root` group; on FreeBSD for example it's known as `wheel`
-/// let find = String::from("root");
-/// // Returns an Option<Result<Group>>, thus the double unwrap. Will panic if there's no `root`
-/// // group.
-/// let res = Group::query( GroupQuery::Name(find.clone()) ).unwrap().unwrap();
-/// assert!(res.name == find);
+// Disable this test on FreeBSD
```

Comment:
Disable it why?  The comment should say.

Reply:
Improved it.